### ISSUE TITLE
Improve app performance

### ISSUE THREAD

**Issue by bbovenzi** [hackforla/food-oasis#748]
### Overview
Currently, there has not been much focus on the actual performance of the app. This matters since we expect many users will access our app on budget phones with slow connections or on old computers.

### Actions

- [x] Audit the current performance of our app
- [x] Discuss possible solutions according to the audit
  - [x] Invite dev team to this issue to review the Lighthouse results and see if there are easy ways to improve performance (i.e. remove unused javascript)
  - [x] Create issue to do / correct ARIA (accessibility) tagging at some point to better support accessibility tools

### Ideas
To get optimal performance may require a full refactor. For now this issue just focuses on low hanging fruit.

### Ressources
See below

**Comment by bbovenzi** [comment]
Map view:
<img width="568" alt="Screen Shot 2020-10-20 at 12 06 28 PM" src="https://user-images.githubusercontent.com/4600967/96668480-dd714000-1320-11eb-8cc6-e221b844a138.png">
Landing page:
<img width="582" alt="Screen Shot 2020-10-20 at 12 12 24 PM" src="https://user-images.githubusercontent.com/4600967/96668481-dea26d00-1320-11eb-945e-12fc17fc400d.png">

**Comment by bbovenzi** [comment]
Might be worth checking out ways to improve map rendering like `reusemaps` https://visgl.github.io/react-map-gl/docs/api-reference/static-map#reusemaps

**Comment by ExperimentsInHonesty** [comment]
Hi @bbovenzi, what tool did you use to test performance.  Please provide link

**Comment by ExperimentsInHonesty** [comment]
@Benbaillou will discuss size with John before prioritizing with Bonnie

**Comment by bbovenzi** [comment]
> Hi @bbovenzi, what tool did you use to test performance. Please provide link

Just Lighthouse in chrome: https://developers.google.com/web/tools/lighthouse/

**Comment by Benbaillou** [comment]
@bbovenzi Thank you

**Comment by staceyrebekahscott** [comment]
Moved to Questions/ In Review as performance assessment should be prioritized, but would like a fresh opinion on the process and tools to accomplish this.

**Comment by staceyrebekahscott** [comment]
@entrotech What is your opinion on how critical this is and how should we be conducting performance evaluations?

**Comment by staceyrebekahscott** [comment]
Moving this to the Waiting column. I would like to consider this in a more cohesive way, as there are some separate issues regarding performance and security that should be built into a more strategic process.

**Comment by staceyrebekahscott** [comment]
@jnaito777 Can you review this issue and advise how this might be related to #928? These issues sound similar and I am wondering if the scopes can be combined. If so, I would say we should close this issue and add the tasks from here onto issue #928. Let me know what you think.

**Comment by fancyham** [comment]
@jnaito777 if you have time, could you do a try at using Google Lighthouse test on the two screens — landing page, map view (not quite sure which page they mean — this is why process documentation is so important!) — and see what our current results are we we can compare vs those 2020 results?

We can treat those earlier ones as the initial benchmark. Maybe we’ve made some significant improvements. 

I think we made some significant map improvements since then, and also reduced the size of the landing page image.

**Comment by jnaito777** [comment]
Here are a few screenshots from Google lighthouse's report viewer. Please let me know if any details need to be added.
Landing page:
<img width="843" alt="Screen Shot 2022-10-13 at 4 23 16 PM" src="https://user-images.githubusercontent.com/114040198/195728671-4df4624e-5227-429b-8088-4d58a05707ed.png">
<img width="793" alt="Screen Shot 2022-10-13 at 4 23 36 PM" src="https://user-images.githubusercontent.com/114040198/195728676-f7e6e496-1723-40c2-aede-574ceb6470e7.png">
<img width="832" alt="Screen Shot 2022-10-13 at 4 29 05 PM" src="https://user-images.githubusercontent.com/114040198/195729180-54ff60bc-c046-42a6-9a4c-e6c5c5ffe6b4.png">

Map view page:
<img width="798" alt="Screen Shot 2022-10-13 at 4 26 29 PM" src="https://user-images.githubusercontent.com/114040198/195729627-37ce9899-7242-42e2-9e73-96be059ddf32.png">
<img width="802" alt="Screen Shot 2022-10-13 at 4 26 53 PM" src="https://user-images.githubusercontent.com/114040198/195729648-a55006dc-07c1-4df7-a249-8502b15c52ee.png">
<img width="793" alt="Screen Shot 2022-10-13 at 4 26 37 PM" src="https://user-images.githubusercontent.com/114040198/195729655-3390f8d0-c3e0-45de-a9ee-528a31f9a25a.png">

From a quick glance, it seems we have made strides in accessibility and best practices. Thoughts?

**Comment by fancyham** [comment]
Wow — nice to see accessibility scores go up (I wonder what changed) but wow, our performance dropped.

Thanks for posting the full details — that’s really helpful.

I think this could be a good time to ask the dev team to take a look at this report (or to run it themselves) and see if they have recommendations for ways to improve the performance?  

Sounds like we have some scripts that aren’t being used?  Also, wondering what might have made our scores go down… it’s hard to tell because the first Lighthouse test doesn’t describe what was looked up or on what size device.

- [ ] Invite dev team to this issue (or create a new one) to review the Lighthouse results and see if there are easy ways to improve performance (i.e. remove unused javascript as recommended by Lighthouse)

For the accessibility ([the ARIA stuff](http://web-accessibility.carnegiemuseums.org/foundations/aria/)), that could be a second issue?

- [ ] Create issue to do ARIA tagging at some point to better support accessibility tools

**Comment by staceyrebekahscott** [comment]
Moved issue to In Progress, as it is being actively worked on. 

@jnaito777 I would recommend attending the Dev meeting (Mondays at 6:30PM) to discuss this issue with them, to follow up on Bryan's suggestions. We give them a heads up on the fola-devs Slack channel when we plan to attend.

**Comment by jnaito777** [comment]
I will try to attend next week's meeting and update here.

**Comment by staceyrebekahscott** [comment]
@jnaito777 A reminder to attend the Dev meeting tonight if you are available to do so.

**Comment by jnaito777** [comment]
After meeting with the dev team, this issue should be broken up into 2 issues.
1. Improving performance-removing unused JavaScript
2. Accessibility-ARIA fix

It sounds like removing unused JavaScript would be difficult (due to how JavaScript is bundled in React) and would take a significant amount of time. However, the ARIA fix should be relatively quick. 

That being said, I will create a new ARIA issue. I will leave it up to John @entrotech to decide if he wants to create a new issue or what to do with this issue.

**Comment by fancyham** [comment]
Created follow-up issues to improve website performance and to make website more accessible for screen readers  based on results from this app performance review:
- #1396 
- #1397
